Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerRoleStore.cs

```diff
@@ -233,53 +239,60 @@ private async Task<EntityModels.Role> GetEntityModel(Guid id)
-                .Include(r => r.GroupRoles)
-                .ThenInclude(gr => gr.Group)
-                .ThenInclude(g => g.GroupRoles)
-                .ThenInclude(gr => gr.Role)
-                .ThenInclude(r => r.SecurableItem)
-                .Include(r => r.RoleUsers)
-                .ThenInclude(ru => ru.User)
-                .ThenInclude(u => u.RoleUsers)
-                .ThenInclude(ur => ur.Role)
-                .ThenInclude(r => r.SecurableItem)
-                .Include(r => r.RoleUsers)
-                .ThenInclude(ru => ru.User)
-                .ThenInclude(u => u.GroupUsers)
-                .ThenInclude(gu => gu.Group)
-                .AsNoTracking()
-                .Where(r => !r.IsDeleted);
-
-            if (!string.IsNullOrEmpty(grain))
+            try
             {
-                roles = roles.Where(r => r.Grain == grain);
-            }
+                var roles = AuthorizationDbContext.Roles
+                    .Include(r => r.SecurableItem)
+                    .Include(r => r.ParentRole)
+                    .Include(r => r.ChildRoles)
+                    .Include(r => r.RolePermissions)
+                    .ThenInclude(rp => rp.Permission)
+                    .ThenInclude(p => p.SecurableItem)
+                    .Include(r => r.GroupRoles)
+                    .ThenInclude(gr => gr.Group)
+                    .ThenInclude(g => g.GroupRoles)
+                    .ThenInclude(gr => gr.Role)
+                    .ThenInclude(r => r.SecurableItem)
+                    .Include(r => r.RoleUsers)
+                    .ThenInclude(ru => ru.User)
+                    .ThenInclude(u => u.RoleUsers)
+                    .ThenInclude(ur => ur.Role)
+                    .ThenInclude(r => r.SecurableItem)
+                    .Include(r => r.RoleUsers)
+                    .ThenInclude(ru => ru.User)
+                    .ThenInclude(u => u.GroupUsers)
+                    .ThenInclude(gu => gu.Group)
+                    .AsNoTracking()
+                    .Where(r => !r.IsDeleted);
+
+                if (!string.IsNullOrEmpty(grain))
+                {
+                    roles = roles.Where(r => r.Grain == grain);
+                }
 
-            if (!string.IsNullOrWhiteSpace(securableItem))
-            {
-                roles = roles.Where(r => r.SecurableItem.Name == securableItem);
-            }
+                if (!string.IsNullOrWhiteSpace(securableItem))
+                {
+                    roles = roles.Where(r => r.SecurableItem.Name == securableItem);
+                }
 
-            if (!string.IsNullOrWhiteSpace(roleName))
-            {
-                roles = roles.Where(r => r.Name == roleName);
-            }
+                if (!string.IsNullOrWhiteSpace(roleName))
+                {
+                    roles = roles.Where(r => r.Name == roleName);
+                }
 
-            foreach (var role in roles)
+                foreach (var role in roles)
```

Comment:
I know this was here before, but this loop over an IQueryable with ToLists inside of it seems to be a performance red flag...we don't have to fix it with this PR, but perhaps we need to address it in a subsequent PR.